tools/boost_names.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+# Copyright 2017 Niklas Claesson
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import collections
+
+Module = collections.namedtuple('Module', ['dirname', 'name', 'libnames'])
+
+LIBS = 'libs'
+
+manual_map = {
+    'callable_traits': 'Call Traits',
+    'crc': 'CRC',
+    'dll': 'DLL',
+    'gil': 'GIL',
+    'graph_parallel': 'GraphParallel',
+    'icl': 'ICL',
+    'io': 'IO State Savers',
+    'msm': 'Meta State Machine',
+    'mpi': 'MPI',
+    'mpl': 'MPL',
+    'multi_array': 'Multi-Array',
+    'multi_index': 'Multi-Index',
+    'numeric': 'Numeric Conversion',
+    'ptr_container': 'Pointer Container',
+    'poly_collection': 'PolyCollection',
+    'qvm': 'QVM',
+    'throw_exception': 'ThrowException',
+    'tti': 'TTI',
+    'vmd': 'VMD',
+    }
+
+extra = [
+    Module('utility', 'Compressed Pair', []),
+    Module('core', 'Enable If', []),
+    Module('functional', 'Functional/Factory', []),
+    Module('functional', 'Functional/Forward', []),
+    Module('functional', 'Functional/Hash', []),
+    Module('functional', 'Functional/Overloaded Function', []),
+    Module('utility', 'Identity Type', []),
+    Module('utility', 'In Place Factory, Typed In Place Factory', []),
+    Module('numeric', 'Interval', []),
+    Module('math', 'Math Common Factor', []),
+    Module('math', 'Math Octonion', []),
+    Module('math', 'Math Quaternion', []),
+    Module('math', 'Math/Special Functions', []),
+    Module('math', 'Math/Statistical Distributions', []),
+    Module('bind', 'Member Function', []),
+    Module('algorithm', 'Min-Max', []),
+    Module('numeric', 'Odeint', []),
+    Module('utility', 'Operators', []),
+    Module('core', 'Ref', []),
+    Module('utility', 'Result Of', []),
+    Module('algorithm', 'String Algo', []),
+    Module('core', 'Swap', []),
+    Module('', 'Tribool', []),
+    Module('numeric', 'uBLAS', []),
+    Module('utility', 'Value Initialized', []),
+    ]
```

Comment:
__[flake8]__
* [E123] closing bracket does not match indentation of opening bracket's line

<!-- Comment by [SideCI](https://sideci.com) -->